src/common/Common.ts

```diff
@@ -102,6 +123,72 @@ export class CommandLineUtils {
             return defaultValue;
         }
     }
+
+    public static createFlagConfig(
+        type: FlagsConfigType,
+        required: boolean
+    ): FlagsConfig {
+        switch (type) {
+            case FlagsConfigType.ApiLevel:
+                return {
+                    apilevel: flags.string({
+                        char: 'a',
+                        description: messages.getMessage(
+                            'apiLevelFlagDescription'
+                        ),
+                        longDescription: messages.getMessage(
+                            'apiLevelFlagDescription'
+                        ),
+                        required,
+                        validate: CommandLineUtils.validateApiLevelFlag
+                    })
+                };
+            case FlagsConfigType.Platform:
+                return {
+                    platform: flags.string({
+                        char: 'p',
+                        description: messages.getMessage(
+                            'platformFlagDescription'
+                        ),
+                        longDescription: messages.getMessage(
+                            'platformFlagDescription'
+                        ),
+                        required,
+                        validate: CommandLineUtils.validatePlatformFlag
+                    })
+                };
+        }
+    }
+
+    private static validateApiLevelFlag(flag: string): boolean {
+        try {
+            Version.from(flag);
+        } catch (error) {
```

Comment:
Constrain posed by `SfdxFlag`'s validate is that we don't have access to other flag. Here, previously we were checking the platform flag and if the check is running for iOS we skip checking for the version string entirely. We can't do that anymore. So I opted to validate the version string for iOS if it is passed.

While it is true that this is `SfdxFlag`'s `validate`'s constraint I'm also feeling now that our commands are amorphous and highly dependent on flags passed to determine how it behaves differently for iOS or Android. Perhaps an implementation/refactor that we can think of in the future is have 2 set of commands, if we need to, for each iOS and Android so that we can avoid a situation like this from the beginning.

Reply:
Validating the flag's value when provided (regardless of iOS or Android) makes sense to me.